### DISCUSSION TITLE
How to disable lights prior to sleep?

### DISCUSSION THREAD

**Discussion by jumper-d-1981** [Makuna/NeoPixelBus#656] [Q&A]
I asked this on gitter and on issues before realizing that discussions was an option.  Apologies for being all over the place.  

I have an issue where I cannot turn off my pixels before deep_sleep on esp32 and esp32 home chat said to ask here, home assistant people suggested using a software switch but that does not seem to work.  I need some advice.

Essentially, when I deep_sleep my device, the lights stay on.  How would I disable lights prior to sleep?  I tried assigning a software switch to the same GPIO but that did not work, I can see the switch turning on and off as per timing settings but it does not affect the light.  Someone suggested using another pin and jumping a resistor to the data pin not sure if that is correct.  Just need to put everything into as low of a power state as possible...to preserve battery charge for as long as possible.  Thanks.

Thoughts?  Thanks.

**Comment by jumper-d-1981** [comment]
Disregard, gitter convo helped me.  Set up a transistor which seems to do the job real well.  Used a separate pin to pull up the transistor...

**Reply by Makuna** [reply]
What transistor/circuit did you use?  

This comes up rarely, but it is still an interesting topic to capture details of the solution.

**Comment by Makuna** [comment]
I am unsure how much power is consumed by the LEDs when they set to show black.  This would be the first thing you should do.  Note that you will need to wait until the strip is fully shown before putting the device into sleep.  Something like...

```
    strip.ClearTo(0); // universal way to say clear to black
    strip.Show(); // this is asynchronous and will return before the LEDs are even started to be updated
    while (!strip.CanShow()) // wait until the last show call completes and the LEDs are off
    {
        yield();
    }
    // now you can sleep the ESP32
```
You can then measure the power the LEDs use when black and then use a mosfet/transistor on the power wire to turn off.

**Reply by jumper-d-1981** [reply]
Just curious, where do I type strip.ClearTo(0); in a yaml?
Again, I am still very novice at HA...and I am aware I can do custom code stuffs in HA but have yet to do so...

**Reply by Makuna** [reply]
That you would have to ask the ESP Home forum, I just do the NeoPixelBus ;-)

**Reply by jumper-d-1981** [reply]
Alright.  For the record.  Yes.  I am still working on this.
Tried the mosfet, bleed voltage was preventing complete powering down, then discovered the on_boot thing...which actually on or off the light just fine.  something odd within the interval component I was using previously.  

BUT I am still having one issue.  I cannot turn the light on with the effect I want.  Am I doing this correctly?  I would love to have scan or rainbow for my final device...to turn on when the thing boots...

here is the yaml config

**Reply by jumper-d-1981** [reply]
esphome:
  name: esphome-web-302d38
  on_boot:
    priority: 600
    then:
      - light.turn_on:
          id: LED_Strip
          brightness: 50%
          red: 0%
          blue: 100%
          green: 0%
      - delay: 5s
      - light.turn_off: LED_Strip
      - delay: 5s
      - light.turn_on:
          id: LED_Strip
          brightness: 50%
          effect: addressable_rainbow
      - delay: 20s
      - light.turn_off: LED_Strip
      - delay: 5s
      - deep_sleep.enter: deep_sleep_1

deep_sleep:
  id: deep_sleep_1
  run_duration: 60s
  sleep_duration: 30s

light:
  - platform: neopixelbus
    variant: WS2812
    type: GRB
    pin: GPIO25
    num_leds: 18
    name: "LED_Strip"
    id: LED_Strip
    effects:
      - addressable_rainbow:

sensor:
  - platform: dht
    pin: 14
    id: my_dht
    temperature:
      name: "Kitchen Temperature"
    humidity:
      name: "Kitchen Humidity"
    update_interval: 8s

**Reply by jumper-d-1981** [reply]
apologies for the formatting atrocity above.

**Reply by Makuna** [reply]
You may want to jump over to esphome forum to ask that question.  This library doesn't deal in xaml at all.
